kbfssync/semaphore_test.go

```diff
@@ -131,19 +131,19 @@ func TestCancel(t *testing.T) {
 	s := NewSemaphore()
 	require.Equal(t, int64(0), s.countForTest())
 
+	// Do this before spawning the goroutine to avoid a race on
+	// the count field of the result of callAcquire.
```

Comment:
Maybe a more robust way to fix this would be this:

```go
	for {
		onRelease, _ := s.tryAcquire(n)
		if onRelease == nil {
			return count, nil
		}

		select {
		case <-onRelease:
			// Go to the top of the loop.
		case <-ctx.Done():
			s.lock.RLock()
			defer s.lock.RUnlock()
			return s.count, errors.WithStack(ctx.Err())
		}
	}
```

Of course then the returned `count` could actually be bigger than the requested `n`, but it might be a good thing to give a more accurate view of the current state to the caller?